### PR TITLE
feat: add and fetch post from server

### PR DESCRIPTION
**Pull request by ravikumar0403**
## Change log:
---
- Create Post modal
- fetch post from server
- Loaders and dark mode

### CODE REVIEW

File: src/redux/features/modalSlice.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  isVisible: false,
+  selectedPost: null,
+};
+
+const modalSlice = createSlice({
+  name: "modal",
+  initialState,
+  reducers: {
+    showModal: (state, action) => {
+      console.log("here");
```
Comment (yashpatel1797) [inline src/redux/features/modalSlice.js:13]: remove console

File: src/components/PostInputModal.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    e.preventDefault();
+    const post = {
+      content: postContent,
+      images: files[0],
+    };
+    dispatch(addPost(post));
+    dispatch(closeModal());
+  };
+
+  return reactDom.createPortal(
```
Comment (yashpatel1797) [inline src/components/PostInputModal.js:26]: good use of react portals.

File: src/components/PostCard.js [hunk 3]
```diff
@@ -1,28 +1,27 @@
...
-        <p className="py-2">
-          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam vel
-          libero eu nibh ullamcorper blandit.
-        </p>
+        <p className="py-2">{post.content}</p>
       </section>
-      {"src" && (
-        <img className="w-full" src="https://picsum.photos/500/300" alt="" />
+      {post.images.length > 0 && (
+        <img className="w-full" src={post.images[0]} alt="" />
```
Comment (yashpatel1797) [inline src/components/PostCard.js:24]: add value to the `alt` attribute.

**Review by yashpatel1797 — APPROVED** [review]
code looks clean
file structure is good
design looks consistent

**Review by JatuFaique — APPROVED** [review]
Code looks clean 
Good redux architecture implementation
Looks good to merge.

**Pull request merged**